SmartStorePerformance/StoreMgr.js

```diff
@@ -0,0 +1,178 @@
+retrieveSoupEntries = forceUtil.promiser(smartstore.retrieveSoupEntries);
+querySoup = forceUtil.promiser(smartstore.querySoup);
+runSmartQuery = forceUtil.promiser(smartstore.runSmartQuery);
+removeFromSoup = forceUtil.promiser(smartstore.removeFromSoup);
+clearSoup = forceUtil.promiser(smartstore.clearSoup);
+getAllStores = forceUtil.promiser(smartstore.getAllStores);
+getAllGlobalStores = forceUtil.promiser(smartstore.getAllGlobalStores);
+removeStore = forceUtil.promiser(smartstore.removeStore);
+removeAllStores = forceUtil.promiser(smartstore.removeAllStores);
+removeAllGlobalStores = forceUtil.promiser(smartstore.removeAllGlobalStores);
+
+const selectSoup = "X1_Custom_Perf_Select";
+const insertSoup = "X1_Custom_Perf_Insert";
+const updateSoup = "X1_Custom_Perf_Update";
+const randomData = ["Quinten", "Gayle", "Sheridan", "Albina", "Marianne", "Avon", "Cambridgeshire", "Buckinghamshire",
+    "Cambridgeshire", "South Deborah", "faker[parts[0]][parts[1]] is not a function",
+    "ba66e370-be9c-4079-bef2-67bcc085aaa6", "cd5d2363-d428-45fc-ad23-5d47b9411ea4", "amos", "van", "haleigh",
+    "I'll back up the bluetooth FTP feed, that should feed the SAS matrix!",
+    "copying the microchip won't do anything, we need to parse the virtual COM pixel!",
+    "If we navigate the bandwidth, we can get to the SCSI hard drive through the open-source SMTP interface!",
+    "Use the back-end HDD matrix, then you can input the neural monitor!"];
+
+async function selectBenchmark(global, querySpec) {
+    var before = Date.now();
+    return querySoup(false, selectSoup, querySpec)
+        .then((queryResult) => {
+            var after = Date.now();
+            var time = (after - before) / 1000;
+            return time;
+        })
+        .catch((error) => {
+            console.log("\n\nquery error: " + error);
+        });
+}
+
+async function insertBenchmark(numEntries) {
+    return clearSoup(false, insertSoup)
+        .then(() => {
+            const x1data = require('./X1_Custom_Perf.json');
+            var insertData = x1data.records.slice(0, numEntries);
+
+            let before = Date.now();
+            return upsertSoupEntries(false, insertSoup, insertData)
+                .then(() => {
+                    let after = Date.now();
+                    let time = (after - before) / 1000;
+                    return time;
+                });
+        });
+}
+
+async function updateBenchmark(fields, rows) {
+    // TODO
+}
+
+function createSoups() {
+    soupExists(false, selectSoup)
+    .then((exists) => {
+        if(!exists) {
+            createX1Soup(selectSoup);
+            createX1Soup(insertSoup);
+            createX1Soup(updateSoup);
+        }
+    })
+    .catch((error) => { console.log("error checking if soup exists: " + error) });
+}
+
+function createX1Soup(soupName) {
+    registerSoup(false, soupName,
+    [
+        { path:"attributes", type: "json1"},
```

Comment:
Is it realistic to index every single fields in a record?
Typically you would only index field you intent to query by.